libnd4j/include/helpers/impl/EigenValsAndVecs.cpp

```diff
@@ -284,6 +294,33 @@ void EigenValsAndVecs<T>::calcEigenVecs(const NDArray& schurMatrixU) {
     }
 }
 
+template <typename T>
+void EigenValsAndVecs<T>::calcEigenVecs(const NDArray& schurMatrixU) {
+    calcEigenVecs_<T>(schurMatrixU, _Vals, _Vecs);
+}
+
+template <typename T>
+void eig_(const NDArray& input, NDArray& vals,  NDArray& vecs ){ 
+
+    assert(input.rankOf() == 2);
```

Comment:
Mind adding error messages to these? Otherwise it will be hard to debug (especially from java)

Reply:
Its accepted just to leave a comment while using the standard runtime assert . I will expand it to have messages. 

I used assert there because it is used internally by the op and the op is responsible for throwing errors and checking before calling.

Reply:
For sure! I was just mentioning that because anyone at the java level debugging something from c++ won't really want to have to click in to and read c++ source code and to try to reverse engineer what the problem is.

Reply:
I followed the old trick to add a message to the assert. now it will be informative for debug